### PR TITLE
Update consignment query

### PR DESCRIPTION
**Pull request by vimleshtna**
Add file filters in consignment's 'files' field to query the files only for given file Ids

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala [hunk 1]
```diff
@@ -134,7 +136,11 @@ object ConsignmentFields {
       Field(
         "files",
         ListType(FileType),
-        resolve = context => DeferFiles(context.value.consignmentid)
+        arguments = FileFiltersInputArg :: Nil,
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala:139]: The `fileFiltersInputArg` will need to be passed into the `getConsignment` query (line 188) as an optional argument I think, otherwise it won't be accessible by the deferred resolve for getting the files.
...
```
Field("getConsignment", OptionType(ConsignmentType),
      arguments = ConsignmentIdArg :: PaginationInputArg :: FileFiltersInputArg :: Nil,
      ...
    )
...
```

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala [hunk 2]
```diff
@@ -229,6 +229,30 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
       files.size shouldBe 1
   }
 
+  "getFiles" should "return files and file metadata where 'selectedFileIds' filter applied" in withContainers {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:232]: Would reword the test slightly to: `... return files and file metadata for selected files where 'selectedFileIds' ...`

I would also add the directory to the selected files list to show it will bring back both types where the type filter is not applied.

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala [hunk 3]
```diff
@@ -229,6 +229,30 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
...
+      val utils = TestUtils(db)
+      val fileRepository = new FileRepository(db)
+      val consignmentId = UUID.fromString("c6f78fef-704a-46a8-82c0-afa465199e66")
+      setUpFilesAndDirectories(consignmentId, utils)
+
+      val files = fileRepository.getFiles(consignmentId, FileFilters(selectedFileIds = Some(List(fileOneId)))).futureValue
+      files.forall(_._1.fileid == fileOneId) shouldBe true
+  }
+
+  "getFiles" should "return files and file metadata where and 'file' type and 'selectedFileIds' filter applied" in withContainers {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:244]: Would reword the test slightly to: `... return files and file metadata for selected files where 'file' type and 'selectedFileIds' filters are applied`

**Review by TomJKing — COMMENTED** [review]
Looks good. Some minor comments. Also the `ConsignmentRouteSpec` tests should be updated with some tests with using the "file filters" you've added.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala (hunk 1 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala:139]: Actually, I tested and it works fine. I think we don't need to pass with `getConsignment` 2field as we are using the `fileFiltersInputArg` in the `files` field only

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala (hunk 2 above)
Comment (vimleshtna) [reply src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:232]: Sure, I'll add it too. Thanks

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala (hunk 1 above)
Comment (TomJKing) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala:139]: As per our conversation, yes correct. Please ignore this comment.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.
